treeherder/model/management/commands/cycle_data.py

```diff
@@ -73,28 +84,192 @@ def prune(id_name, model):
 
 
 class PerfherderCycler(DataCycler):
-    source = PERFHERDER.title()
-    max_runtime = datetime.timedelta(hours=23)
-
-    def __init__(self, days, chunk_size, sleep_time, is_debug=None, logger=None, **kwargs):
-        super().__init__(days, chunk_size, sleep_time, is_debug, logger)
-        if (
-            days < MINIMUM_PERFHERDER_EXPIRE_INTERVAL
-            and settings.SITE_HOSTNAME != 'treeherder-prototype2.herokuapp.com'
-        ):
-            raise ValueError(
-                'Cannot remove performance data that is more recent than {} days'.format(
-                    MINIMUM_PERFHERDER_EXPIRE_INTERVAL
-                )
-            )
+    DEFAULT_MAX_RUNTIME = timedelta(hours=23)
+
+    def __init__(
+        self,
+        chunk_size: int,
+        sleep_time: int,
+        is_debug: bool = None,
+        max_runtime: timedelta = None,
+        strategies: List[RemovalStrategy] = None,
+        **kwargs,
+    ):
+        super().__init__(chunk_size, sleep_time, is_debug)
+        self.started_at = None
+        self.max_runtime = max_runtime or PerfherderCycler.DEFAULT_MAX_RUNTIME
+        self.strategies = strategies or RemovalStrategy.fabricate_all_strategies(chunk_size)
 
     def cycle(self):
```

Comment:
This function describes **how** we are deleting the data, which might not be a straightforward naming. I would describe **what** it deletes, like:
`def delete_older_than(self, timedelta=DEFAULT_MAX_RUNTIME)`
and let the user decide whether he wants to use the default or some of it's own.
But this is just a recommendation.

Reply:
I disagree. The `PerfherderCycler` doesn't really know what it deletes. That's the responsibility of the strategies it encompasses.
And the strategies don't really stick to the idea of deleting something that's older than X time. The strategies employ various garbage collecting techniques which can be pretty different.